src/include/duckdb/function/aggregate/minmax_n_helpers.hpp

```diff
@@ -102,20 +102,23 @@ class UnaryAggregateHeap {
 public:
 	UnaryAggregateHeap() = default;
 
-	explicit UnaryAggregateHeap(idx_t capacity_p) : capacity(capacity_p) {
-		heap.reserve(capacity);
+	UnaryAggregateHeap(ArenaAllocator &allocator, idx_t capacity_p) {
+		Initialize(allocator, capacity_p);
 	}
 
-	void Initialize(const idx_t capacity_p) {
+	void Initialize(ArenaAllocator &allocator, const idx_t capacity_p) {
 		capacity = capacity_p;
-		heap.reserve(capacity);
+		auto ptr = allocator.AllocateAligned(capacity * sizeof(HeapEntry<T>));
+		memset(ptr, 0, capacity * sizeof(HeapEntry<T>));
+		heap = reinterpret_cast<HeapEntry<T> *>(ptr);
```

Comment:
Im not sure if this is UB in the case of `HeapEntry<string_t>` which is not a trivial type. I _think_ you need to loop over the allocated memory and placement new entries at each slot? (AFAIK you don't want to do placement array new)

Reply:
I think `string_t` is a trivial type as it's only composed of trivial types and doesn't have a destructor - memsetting it to 0 does the "right thing" intuitively as well:

```

	union {
		struct {
			uint32_t length;
			char prefix[4];
			char *ptr;
		} pointer;
		struct {
			uint32_t length;
			char inlined[12];
		} inlined;
	} value;
```

-> sets length to 0, and inlined string data to `\0`

Reply:
Hmm, I guess, you're right when I checked before it probably wasn't fully defined
![image](https://github.com/user-attachments/assets/28b69e97-16af-43cc-8c60-0551ace53daf)

Although, `HeapEntry<string_t>` doesn't seem to be

Reply:
But it is standard layout, so I think its ok?

![image](https://github.com/user-attachments/assets/bbe71865-140a-4a7b-95e9-fac7daf89ca7)

Reply:
Ah yeah, `HeapEntry<string_t>` is specialized and isn't copyable/doesn't have a trivial move constructor. It should still be fine to `memset` to zero though:

```cpp

struct HeapEntry<string_t> {
	string_t value;
	uint32_t capacity;
	data_ptr_t allocated_data;
}
```